src/CanvasRenderingContext2d.cc

```diff
@@ -1479,330 +1492,325 @@ NAN_GETTER(Context2d::GetGlobalCompositeOperation) {
+      {"luminosity", CAIRO_OPERATOR_HSL_LUMINOSITY},
+      // non-standard:
+      {"saturate", CAIRO_OPERATOR_SATURATE}
+    };
+    auto op = blendmodes.find(opStr.Utf8Value());
+    if (op != blendmodes.end()) cairo_set_operator(ctx, op->second);
+  }
 }
 
 /*
  * Get shadow offset x.
  */
 
-NAN_GETTER(Context2d::GetShadowOffsetX) {
-  CHECK_RECEIVER(Context2d.GetShadowOffsetX);
-  Context2d *context = Nan::ObjectWrap::Unwrap<Context2d>(info.This());
-  info.GetReturnValue().Set(Nan::New<Number>(context->state->shadowOffsetX));
+Napi::Value
+Context2d::GetShadowOffsetX(const Napi::CallbackInfo& info) {
+  return Napi::Number::New(env, state->shadowOffsetX);
 }
 
 /*
  * Set shadow offset x.
  */
 
-NAN_SETTER(Context2d::SetShadowOffsetX) {
-  CHECK_RECEIVER(Context2d.SetShadowOffsetX);
-  Context2d *context = Nan::ObjectWrap::Unwrap<Context2d>(info.This());
-  context->state->shadowOffsetX = Nan::To<double>(value).FromMaybe(0);
+void
+Context2d::SetShadowOffsetX(const Napi::CallbackInfo& info, const Napi::Value& value) {
+  Napi::Number numberValue;
+  if (value.ToNumber().UnwrapTo(&numberValue)) state->shadowOffsetX = numberValue.DoubleValue();
 }
 
 /*
  * Get shadow offset y.
  */
 
-NAN_GETTER(Context2d::GetShadowOffsetY) {
-  CHECK_RECEIVER(Context2d.GetShadowOffsetY);
-  Context2d *context = Nan::ObjectWrap::Unwrap<Context2d>(info.This());
-  info.GetReturnValue().Set(Nan::New<Number>(context->state->shadowOffsetY));
+Napi::Value
+Context2d::GetShadowOffsetY(const Napi::CallbackInfo& info) {
+  return Napi::Number::New(env, state->shadowOffsetY);
 }
 
 /*
  * Set shadow offset y.
  */
 
-NAN_SETTER(Context2d::SetShadowOffsetY) {
-  CHECK_RECEIVER(Context2d.SetShadowOffsetY);
-  Context2d *context = Nan::ObjectWrap::Unwrap<Context2d>(info.This());
-  context->state->shadowOffsetY = Nan::To<double>(value).FromMaybe(0);
+void
+Context2d::SetShadowOffsetY(const Napi::CallbackInfo& info, const Napi::Value& value) {
+  Napi::Number numberValue;
+  if (value.ToNumber().UnwrapTo(&numberValue)) state->shadowOffsetY = numberValue.DoubleValue();
 }
 
 /*
  * Get shadow blur.
  */
 
-NAN_GETTER(Context2d::GetShadowBlur) {
-  CHECK_RECEIVER(Context2d.GetShadowBlur);
-  Context2d *context = Nan::ObjectWrap::Unwrap<Context2d>(info.This());
-  info.GetReturnValue().Set(Nan::New<Number>(context->state->shadowBlur));
+Napi::Value
+Context2d::GetShadowBlur(const Napi::CallbackInfo& info) {
+  return Napi::Number::New(env, state->shadowBlur);
 }
 
 /*
  * Set shadow blur.
  */
 
-NAN_SETTER(Context2d::SetShadowBlur) {
-  CHECK_RECEIVER(Context2d.SetShadowBlur);
-  int n = Nan::To<double>(value).FromMaybe(0);
-  if (n >= 0) {
-    Context2d *context = Nan::ObjectWrap::Unwrap<Context2d>(info.This());
-    context->state->shadowBlur = n;
-  }
+void
+Context2d::SetShadowBlur(const Napi::CallbackInfo& info, const Napi::Value& value) {
+  Napi::Number n;
+  if (value.ToNumber().UnwrapTo(&n) && n.DoubleValue() >= 0) state->shadowBlur = n.DoubleValue();
 }
```

Comment:
```
  78) WPT: shadows
       2d.shadow.attributes.shadowBlur.invalid:

      AssertionError [ERR_ASSERTION]: ctx.shadowBlur
      + expected - actual

      --2147483648
      +1
      
      at Context.<anonymous> (test\wpt\generated\shadows.js:130:10)
```

```
ctx.shadowBlur = 1;
ctx.shadowBlur = Infinity;
assert.strictEqual(ctx.shadowBlur, 1, "ctx.shadowBlur", "1")
```

Converting any floating type value to an integer, where the value can't be represented by the integer, is undefined behavior. That might be why you're seeing it fail on master and I'm not.

The canvas spec says:

> context.[shadowBlur](https://html.spec.whatwg.org/multipage/canvas.html#dom-context-2d-shadowblur) [ = value ]
>
>    Can be set, to change the blur level. **Values that are not finite numbers greater than or equal to zero are ignored.**

This would work:

```cpp
  Napi::Number n;
  if (value.ToNumber().UnwrapTo(&n)) {
    double v = n.DoubleValue();
    if (v >= 0 && v <= std::numeric_limits<decltype(state->shadowBlur)>::max()) {
      state->shadowBlur = v;
    }
  }
```

(Using `Int32Value()` does *not* work; that returns 0.)